metals/src/main/scala/scala/meta/internal/metals/BuildTargetInfo.scala

```diff
@@ -100,7 +101,10 @@ class BuildTargetInfo(buildTargets: BuildTargets) {
       )
     })
     commonInfo.foreach(info => {
-      output ++= getSection("Base Directory", List(info.baseDirectory))
+      output ++= getSection(
+        "Base Directory",
+        List(URLDecoder.decode(info.baseDirectory, "UTF-8"))
```

Comment:
Calling `URLDecoder.decode` on strings that may not be encoded is dangerous. Here, `baseDirectory` can contain spaces and it will cause `URLDecoder` to throw an exception.

Probably we should write our own `URIDecoder`

Reply:
Same case as in https://github.com/scalameta/metals/pull/3701

Reply:
I created a separate decode and encoder for URIs, it's probably not perfect, but seems to work fine for the current cases.